### ISSUE TITLE
Add modelingtoolkitize support

### ISSUE THREAD

**Issue by jonniedie** [jonniedie/ComponentArrays.jl#16]
I don't know how much work it would be, but it would be really cool to have a `modelingtoolkitize` overload for `ODEProblem`s of `ComponentArray`s that takes the property names of the `ComponentArray` and assigns them to `ODESystem` and variable names. Just by quick inspection, this doesn't seem too difficult.

**Comment by bgroenks96** [comment]
I'm not sure if this is what you're referring to or not, but it seems that `ComponentArrays` is completely incompatible with `modelingtoolkitize`. For example, if my parameter vector `p` is a `ComponentVector`:

```
p_true = ComponentArray(k=0.42)
jac_proto = Tridiagonal(similar(u0,nknots-1), similar(u0), similar(u0, nknots-1))
prob = ODEProblem(ODEFunction(f,jac_prototype=jac_proto), u0, (0.0,1.0), p_true)
```

Then `modelingtoolkitize` fails

```
sys = modelingtoolkitize(prob)
```

with `type Array has no field k`

**Comment by bgroenks96** [comment]
I can create a separate issue, if appropriate.

**Comment by jonniedie** [comment]
Yep, that's what I mean. The snag I ran into is that I haven't found a good way to hook into the internals of the function and let ModelingToolkit know that `p_true.k` actually means `p[1]`. It would be an easy problem to solve, I think, if [`ODESystem` was allowed to hold more generic arrays](https://github.com/SciML/ModelingToolkit.jl/blob/ed2de3e2cdf4f99b29786627588bf5985c169c90/src/systems/diffeqs/odesystem.jl#L31) because then I wouldn't need to hook into the function at all. I could just pass a `ComponentArray` of ModelingToolkit variables through the function and it should just work, I think. I'll have to dig in a little further to see if there is a good reason that the states and parameters of `ODESystem` have to be plain `Vector`s or if that can be relaxed.
